### DISCUSSION TITLE
What about a CLI based mechanism to create / modify users

### DISCUSSION THREAD

**Discussion by brucealthompson** [hrfee/jfa-go#67] [Ideas]
While this project is quite good, its quite a bit more than I need. I just need a CLI based mechanism to create / modify users in Jellyfin. What I want to do is automate the initial Jellyfin setup process based on user information I obtain through other mechanisms.

What I was thinking was I could create an initial user with no password just by modifying the Jellyfin SQL database directlt. From there, I believe I could use the JF API to add other config information such as user password and other parameters.

I think I could use mediabrowser.go and jfapi.go to implement the jf API and put a CLI wrapper around this code. Do you think this is a reaosnable approach to implementing a basic JF initialization CLI or would you recommend a different appproach?

**Comment by hrfee** [comment]
Yeah, a little cli client should be pretty easy with the mediabrowser part of the project. You should be able to just `go get github.com/hrfee/jfa-go/mediabrowser` to use it. Working directly with jellyfin API might also be fun too, once you get over the hurdle of authentication it's pretty simple (also heard they finally added proper docs for it in 10.7.0).

**Reply by brucealthompson** [reply]
OK. I've started this but not done yet. Got basic jellyfin authentication working. Now I need to add a couple commands (like add user). I think this is going to work. I'll give you the code if you want it.

**Reply by brucealthompson** [reply]
I have finished the app. I can add and delete users via the app cli. The most difficult part for me was creating the 1st admin user. I had to configure a new jellyfin install with a well known admin user and then copy jellyfin.db to bring that user to a new jellyfin install.
In any case, I have the code now. You can have it if you want it.
